co-log/example/1-intro/Intro.md

```diff
@@ -0,0 +1,109 @@
+# Intro: Using `LogAction`
+
+This tutorial is an introduction to `co-log`. It contains basic examples of
+using core data types and functions.
+
+You can run this tutorial by calling the following command:
+
+```shell
+cabal new-run intro
+```
+
+## Preamble: imports and language extensions
+
+Since this is a literate haskell file, we need to specify all our language
+extensions and imports up front.
+
+```haskell
+import Colog.Core (LogAction (..), (<&), logStringStdout)
+```
+
+## Core data type
+
+`co-log` is based on the following data type:
+
+```idris
+newtype LogAction m msg = LogAction
+    { unLogAction :: msg -> m ()
+    }
+```
+
+Logging action is a function from some message of user-defined type `msg` that
+performs all logic inside some monad `m`. In the `co-log` library **logger** is represented as a
+value. With such approach, you can modify the way you do logging by simply performing some
+transformations with the value you have.
+
+Let's first look at a very basic example of using just `putStrLn` for logging:
+
+```haskell
+example0 :: IO ()
+example0 = do
+    putStrLn "Example 0: First message"
+    putStrLn "Example 0: Second message"
+```
+
+Using `putStrLn` for logging is a very simple and basic approach for logging.
+When your application becomes bigger and more complex, you might want to bring
+some logging library into it.
```

Comment:
probably add reasons why would you want that, like, to let the library handle more difficult cases or anything like that?